docs/system_overview/ssl_and_tls.md

```diff
@@ -0,0 +1,148 @@
+
+To generate a Certificate Signing Request that can be sent to a CA, run the 'openssl req' command using the previously generated key:
+
+~~~
+irods@hostname:~/ $ openssl req -new -key server.key -out server.csr
+~~~
+
+To generate a self-signed certificate, also run 'openssl req', but with slightly different parameters. In the openssl command, you can put as many days as you wish:
+
+~~~
+irods@hostname:~/ $ openssl req -new -x509 -key server.key -out server.crt -days 365
+~~~
+
+### Create the certificate chain file
+
+If you are using a self-signed certificate, the chain file is just the same as the file with the certificate (server.crt).  If you have received a certificate from a CA, this file contains all the certificates that together can be used to verify the certificate, from the host certificate through the chain of intermediate CAs to the ultimate root CA.
+
+An example best illustrates how to create this file. A certificate for a host 'irods.example.org' is requested from the proper domain registrar. Three files are received from the CA: irods.crt, PositiveSSLCA2.crt and AddTrustExternalCARoot.crt. The certificates have the following 'subjects' and 'issuers':
+
+~~~
+openssl x509 -noout -subject -issuer -in irods.crt
+subject= /OU=Domain Control Validated/OU=PositiveSSL/CN=irods.example.org
+issuer= /C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=PositiveSSL CA 2
+openssl x509 -noout -subject -issuer -in PositiveSSLCA2.crt
+subject= /C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=PositiveSSL CA 2
+issuer= /C=SE/O=AddTrust AB/OU=AddTrust External TTP Network/CN=AddTrust External CA Root
+openssl x509 -noout -subject -issuer -in AddTrustExternalCARoot.crt
+subject= /C=SE/O=AddTrust AB/OU=AddTrust External TTP Network/CN=AddTrust External CA Root
+issuer= /C=SE/O=AddTrust AB/OU=AddTrust External TTP Network/CN=AddTrust External CA Root
+~~~
+
+The irods.example.org cert was signed by the PositiveSSL CA 2, and that the PositiveSSL CA 2 cert was signed by the AddTrust External CA Root, and that the AddTrust External CA Root cert was self-signed, indicating that it is the root CA (and the end of the chain).
+
+To create the chain file for irods.example.org:
+
+~~~
+irods@hostname:~/ $ cat irods.crt PositiveSSLCA2.crt AddTrustExternalCARoot.crt > chain.pem
+~~~
+
+### Generate OpenSSL parameters
+
+Generate some Diffie-Hellman parameters for OpenSSL:
+
+~~~
+irods@hostname:~/ $ openssl dhparam -2 -out dhparams.pem 2048
```

Comment:
Is 2048 still good enough? Then again, that's for the admin to figure out.

Reply:
According to the OpenSSL documentation (https://docs.openssl.org/3.0/man1/openssl-dhparam), the `-2` generator has a default `numbits` of 2048, so we technically don't even need that in this instruction. Regardless, I agree that it's up to the admin.

Reply:
I say we leave it alone. I was not able to find anything online which indicated that 2048 was insufficient. All I know is that 10000 is the highest, and 2048 is sufficient for our currently supported platforms (see https://github.com/irods/irods/issues/6414).